src/Data/Fin/Relation/Unary/Top.agda

```diff
@@ -0,0 +1,257 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- The 'top' view of Fin
+--
+-- This is an example of a view of (elements of) a datatype,
+-- here i : Fin (suc n), which exhibits every such i as
+-- * either, i = fromℕ n
+-- * or, i = inject₁ j for a unique j : Fin n
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+module Data.Fin.Relation.Unary.Top where
+
+open import Data.Empty using (⊥; ⊥-elim)
+open import Data.Nat.Base using (ℕ; zero; suc; _<_; _∸_)
+open import Data.Nat.Properties using (n∸n≡0; +-∸-assoc)
+open import Data.Fin.Base using (Fin; zero; suc; toℕ; fromℕ; fromℕ<; inject₁)
+open import Data.Fin.Properties as Fin
+  using (suc-injective; inject₁-injective; toℕ-fromℕ; toℕ<n; toℕ-inject₁; inject₁ℕ<)
+open import Relation.Binary.PropositionalEquality
+
+private
+  variable
+    n : ℕ
+
+------------------------------------------------------------------------
+-- The View, considered as a unary relation on Fin (suc n)
+
+-- First, a lemma not in `Data.Fin.Properties`,
```

Comment:
The lemma really ought to be moved to its proper home...

Reply:
[Ooops. Hit the wrong button]

I agree! But as with your subsequent comment, it's really a case of "the library used to define things a certain way, and that meant that this property appears only modulo `toℕ ` --- but that's injective, so... meh".

But I did like putting it here for didactic reasons: it's the lemma which tells the user (if not,alas, the typechecker) that the view is a *disjoint* cover of `Fin (suc n)`. So I am open to suggestions as to renaming, or moving. But moving goes with a  more general weeding out of stuff from `Data.Fin` I think.

Reply:
I'll do two things here:
* move to `Data.Fin.Properties`
* rename to `fromℕ≢inject₁` (murder your darlings)